core/src/main/java/io/questdb/griffin/engine/functions/catalogue/TableListFunctionFactory.java

```diff
@@ -126,6 +126,7 @@ private class TableListRecordCursor implements RecordCursor {
             @Override
             public void close() {
                 findPtr = ff.findClose(findPtr);
+                metaReader.clear();//release FD of last table on the list
```

Comment:
nit: we could call `close()` here instead of introducing `clear()` method since it has more or less the same effect.